tests/integration/e2e/test_basic_flow.py

```diff
@@ -70,22 +140,47 @@ async def test_test_app_actually_set_up(ops_test: OpsTest, deploy_test_app):
         apps=[producer], status="active", timeout=1000, idle_period=40
     )
 
+    if integrator:
+        # start producer process on new units
+        assert producer_parameters
+        pid_1 = await fetch_action_start_process(
+            ops_test.model.applications[producer].units[1], producer_parameters
+        )
+        logger.info(f"Producer process started with pid: {pid_1}")
+        pid_2 = await fetch_action_start_process(
+            ops_test.model.applications[producer].units[2], producer_parameters
+        )
+        logger.info(f"Producer process started with pid: {pid_2}")
+
     await asyncio.sleep(100)
 
-    # scale up consumer
     logger.info("Scale up consumer")
     await ops_test.model.applications[consumer].add_units(count=2)
     await ops_test.model.block_until(lambda: len(ops_test.model.applications[consumer].units) == 3)
     await ops_test.model.wait_for_idle(
         apps=[consumer], status="active", timeout=1000, idle_period=40
     )
 
+    if integrator:
+        # start consumer process on new units
+        assert consumer_parameters
+        pid_1 = await fetch_action_start_process(
+            ops_test.model.applications[consumer].units[1], consumer_parameters
+        )
+        logger.info(f"Consumer process started with pid: {pid_1}")
+        pid_2 = await fetch_action_start_process(
+            ops_test.model.applications[consumer].units[2], consumer_parameters
+        )
+        logger.info(f"Consumer process started with pid: {pid_2}")
+
     await asyncio.sleep(100)
 
     logger.info("Scale down consumer")
     await ops_test.model.applications[consumer].destroy_units(f"{consumer}/2")
     await ops_test.model.applications[consumer].destroy_units(f"{consumer}/1")
-    await ops_test.model.block_until(lambda: len(ops_test.model.applications[consumer].units) == 1)
+    await ops_test.model.block_until(
+        lambda: len(ops_test.model.applications[consumer].units) == 1, timeout=1200
+    )
     await ops_test.model.wait_for_idle(apps=[consumer], status="active", timeout=1000)
```

Comment:
**nitpick/suggestion:** You can add `wait_for_exact_units=1` as an arg to the following `wait_for_idle`